neurodsp/filt.py

```diff
@@ -69,204 +68,215 @@ def filter_signal(sig, fs, pass_type, fc, n_cycles=3, n_seconds=None,
         Filter kernel. Only returned if 'return_kernel' is True.
     """
 
-    # Check that frequency cutoff inputs are appropriate
-    if pass_type in ('bandpass', 'bandstop'):
-        # Check, if fc is a tuple and performing bandpass/stop, that
-        #   the second cutoff frequency is greater than the first
-        if isinstance(fc, tuple):
-            if fc[0] >= fc[1]:
-                raise ValueError('Second cutoff frequency must be greater than first.')
+    # Check inputs & compute the nyquist frequency
+    f_lo, f_hi = check_filter_definitions(pass_type, fc)
+    f_nyq = fs / 2.
 
-        if isinstance(fc, (int, float)):
-            raise ValueError('Two cutoff frequencies required for bandpass and bandstop filters')
-        if len(fc) != 2:
+    # Remove any NaN on the edges of 'sig'
+    first_nonan = np.where(~np.isnan(sig))[0][0]
+    last_nonan = np.where(~np.isnan(sig))[0][-1] + 1
+    sig_old = np.copy(sig)
+    sig = sig[first_nonan:last_nonan]
+
+    if iir:
+        iir_checks(pass_type, n_seconds, butterworth_order, remove_edge_artifacts, verbose)
+        sig_filt, b_vals, a_vals = iir_filt(sig, pass_type, f_lo, f_hi, f_nyq,
+                                            butterworth_order, plot_freq_response)
+    else:
+        sig_filt, filt_len, kernel = fir_filt(sig, fs, pass_type, f_lo, f_hi, f_nyq, n_cycles,
+                                              n_seconds, plot_freq_response)
+
+    # Compute transition bandwidth
+    if compute_transition_band and verbose:
+        if not iir:
+            a_vals, b_vals = 1, kernel
+        calc_transition_band(b_vals, a_vals, fs, f_lo, f_hi, f_nyq, pass_type)
+
+    # Remove edge artifacts
+    if not iir and remove_edge_artifacts:
```

Comment:
This doesn't deal with the scenario where `iir = True` and `remove_edge_artifacts = True`. Of course, that's because we can't completely remove edge artifacts from IIR filters. But if the user passes in this input, we should throw a warning like "Edge artifacts are not removed for IIR filters"

Reply:
Hah. Nvm. See that in line 152.